pyproject.toml

```diff
@@ -11,6 +11,19 @@ namespace_packages = true
 explicit_package_bases = true
 exclude = ["thirdparty/"]
 
-[tool.black]
+[tool.ruff]
+extend-exclude = ["thirdparty"]
+extend-include = ["SConstruct", "SCsub"]
 line-length = 120
-extend-exclude = ".*thirdparty/.*"
+target-version = "py37"
+
+[tool.ruff.lint]
+extend-select = [
+	"I", # isort
```

Comment:
I think that also some other usable rule groups could be enabled here in this or other MR - https://docs.astral.sh/ruff/rules/ like:

- https://docs.astral.sh/ruff/rules/#flake8-bugbear-b
- https://docs.astral.sh/ruff/rules/#flake8-simplify-sim
- https://docs.astral.sh/ruff/rules/#perflint-perf
- some lints from https://docs.astral.sh/ruff/rules/#ruff-specific-rules-ruf

Reply:
Let's assess those in a follow-up PR, I think this one is big enough for now :)